test/failure_app_test.rb

```diff
@@ -232,6 +232,33 @@ def call_failure(env_params={})
         assert_equal 'http://test.host/users/sign_in.js', @response.second["Location"]
       end
     end
+
+    test 'returns to configured authentication_failure_route with symbol' do
+      swap Devise, authentication_failure_route: :root_path do
+        call_failure
+        assert_equal 302, @response.first
+        assert_equal 'You need to sign in or sign up before continuing.', @request.flash[:alert]
+        assert_equal 'http://test.host/', @response.second['Location']
+      end
+    end
+
+    test 'returns to root if configured authentication_failure_route not exists' do
+      swap Devise, authentication_failure_route: :unexists do
+        call_failure
+        assert_equal 302, @response.first
+        assert_equal 'You need to sign in or sign up before continuing.', @request.flash[:alert]
+        assert_equal 'http://test.host/', @response.second['Location']
+      end
+    end
+
+    test 'returns to configured authentication_failure_route with lambda' do
+      swap Devise, authentication_failure_route: ->(_warden_options) { :new_user_registration_url } do
+        call_failure
+        assert_equal 302, @response.first
+        assert_equal 'You need to sign in or sign up before continuing.', @request.flash[:alert]
+        assert_equal 'http://test.host/users/sign_up', @response.second['Location']
```

Comment:
is it should be `/sign_in`?

Reply:
@awcodify The `authentication_failure_route` set to a lambda return `:new_user_registration_url`, so `/sign_up` is correct.